src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -143,13 +146,25 @@ private void connect(String username, String password, String hostname, int port
 
     try {
       ExperimenterData user = this.gateway.connect(loginCredentials);
-      this.securityContext = new SecurityContext(user.getGroupId());
+
+      long userGroupID = user.getGroupId();
+
+      if(!group.equalsIgnoreCase("default")){
```

Comment:
I'm not in the know, why do we need to check for a default value here?

Reply:
The idea is to allow the change of the user group (which is kind of a project space in OMERO), so that the connection context can be changed and we can control which set of projects we can see. This is just a bit of functionality to give us flexibility down the line to admin user access.

`user.getGroupId();` gives us the default group for the user, but the user can be part of many user groups, which we get with `user.getGroups()`. However I am open to suggestions :)